### PR TITLE
Fix socketio to work with servers v1.0+

### LINKED ISSUE #13929: socket.io 1.3.6 can't create custom namespace
@himynameschris 
Tried with server code from 
https://github.com/himynameschris/socket.io-testserver

SocketIO::connect("http://localhost:3010/testpoint", *this);

can only connect to default namespace. 

this is the server log
  socket.io:socket socket connected - writing packet +0ms
  socket.io:socket joining room I6LarO9Dg0hqscrpAAAA +12ms
connection
  socket.io:socket joined room I6LarO9Dg0hqscrpAAAA +36ms

and this is the log from cocos2d-x
SocketIOHelper::init called, connecting to url : ws://127.0.0.1:3010/testpoint
SIOClientImpl::init() successful
SIOClientImpl::handshake() called
SIOClientImpl::handshake() waiting
In the constructor of HttpClient!
SIOClientImpl::handshakeResponse() called
handshake completed
response code: 200
SIOClientImpl::handshake() succeeded
SIOClientImpl::handshake() dump data: 97:0{"sid":"S91jNvR2YZpu4ifsAAAI","upgrades":["websocket"],"pingInterval":25000,"pingTimeout":60000}
SIOClientImpl::handshake() Socket.IO 1.x detected
done parsing 1.x
SIOClientImpl::openSocket() called
[WebSocket::init] _host: 127.0.0.1, _port: 3010, _path: /socket.io/1/websocket/?EIO=2&transport=websocket&sid=S91jNvR2YZpu4ifsAAAI
[1442582986:2844] NOTICE: Initial logging level 7
[1442582986:2845] NOTICE: Library version: 1.3 901bc8d
[1442582986:2845] NOTICE: IPV6 compiled in and enabled
[1442582986:2845] NOTICE: libev support not compiled in
[1442582986:2845] NOTICE:  static allocation: 4496 + (16 x 1024 fds) = 20880 bytes
[1442582986:2846] NOTICE:  per-conn mem: 184 + 1594 headers + protocol rx buf
-->SEND:40::/testpoint,:
SIOClientImpl::onOpen socket connected!
SIOClientImpl::onMessage received: 40
Message code: [4]
Socket Connected
SIOClientImpl::onMessage received: 42["message","Simple message test from default namespace"]
Message code: [4]
Event Received (["message","Simple message test from default namespace"])
event name message between 1 and 7
SIOClientImpl::onMessage received: 42["message",{"name":"myname","type":"mytype"}]
Message code: [4]
Event Received (["message",{"name":"myname","type":"mytype"}])
event name message between 1 and 7
SIOClientImpl::onMessage received: 42["testevent",{"name":"myname","type":"mytype"}]
Message code: [4]
Event Received (["testevent",{"name":"myname","type":"mytype"}])
event name testevent between 1 and 9
-->SEND:2:::
Heartbeat sent

### PR DESCRIPTION
**Pull request by himynameschris**
Fixes incorrect packet format by updating the separator between data fields for socketio v1.x packets

fixes #13929

### CODE REVIEW

**Comment by himynameschris** [comment]
also fixes cocos2d-js/#1797

https://github.com/cocos2d/cocos2d-js/issues/1797

File: tests/cpp-tests/Classes/ExtensionsTest/NetworkTest/SocketIOTest.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ void SocketIOTest::closedSocketAction(network::SIOClient* client)
 void SocketIOTest::onMenuSIOClientClicked(cocos2d::Ref *sender)
 {
 	//create a client by using this static method, url does not need to contain the protocol
-	_sioClient = SocketIO::connect("ws://dev.channon.us:3010", *this);
+	_sioClient = SocketIO::connect("ws://localhost:3010", *this);
```
Comment (pandamicro) [inline tests/cpp-tests/Classes/ExtensionsTest/NetworkTest/SocketIOTest.cpp:198]: If we change the test case to use `localhost`, it means we must have a socket io server locally, it's not very friendly for testing. Is this necessary?

**Comment by pandamicro** [comment]
Thank you for the fix !
I want to know why change the SocketIO server in test case to `localhost` before merge it

**Comment by himynameschris** [comment]
my server at dev.channon.us is no longer running, so we can either setup a long term test server somewhere or change it to localhost and expect dev's to have their own server to test with. Testing is as easy as 1) installing nodejs 2) clone the testserver repo 3) npm install 4) node app

**Comment by pandamicro** [comment]
OK, we will find a way to open another server for testing.

@VisualSJ Can your server setup for node js ?

**Comment by man2** [comment]
I think it is also important for developer to know the server code as well as client side code, so setting up server code is also important here.  Just need to find a way for this to be setup easily

**Comment by ZeroHackeR** [comment]
There is still a bug - SocketIO - Connect failed after Disconnected: 
https://github.com/cocos2d/cocos2d-x/issues/14077

**Comment by dumganhar** [comment]
@pandamicro , why was it merged with `localhost`, our QA team will not be able to test this?
How to resolve it ? How to set up the test environment?

**Pull request merged**